token/cli/src/main.rs

```diff
@@ -3071,6 +3181,208 @@ async fn command_enable_disable_confidential_transfers(
     })
 }
 
+#[derive(PartialEq, Eq)]
+enum ConfidentialInstructionType {
+    Deposit,
+    Withdraw,
+}
+
+#[allow(clippy::too_many_arguments)]
+async fn command_deposit_withdraw_confidential_tokens(
+    config: &Config<'_>,
+    token_pubkey: Pubkey,
+    owner: Pubkey,
+    maybe_account: Option<Pubkey>,
+    bulk_signers: BulkSigners,
+    ui_amount: Option<f64>,
+    mint_decimals: Option<u8>,
+    instruction_type: ConfidentialInstructionType,
+    elgamal_keypair: Option<&ElGamalKeypair>,
+    aes_key: Option<&AeKey>,
+) -> CommandResult {
+    if config.sign_only {
+        panic!("Sign-only is not yet supported.");
+    }
+
+    // check if mint decimals provided is consistent
+    let mint_info = config.get_mint_info(&token_pubkey, mint_decimals).await?;
+
+    if !config.sign_only && mint_decimals.is_some() && mint_decimals != Some(mint_info.decimals) {
+        return Err(format!(
+            "Decimals {} was provided, but actual value is {}",
+            mint_decimals.unwrap(),
+            mint_info.decimals
+        )
+        .into());
+    }
+
+    let decimals = if let Some(decimals) = mint_decimals {
+        decimals
+    } else {
+        mint_info.decimals
+    };
+
+    // derive ATA if account address not provided
+    let token_account_address = if let Some(account) = maybe_account {
+        account
+    } else {
+        let token = token_client_from_config(config, &token_pubkey, Some(decimals))?;
+        token.get_associated_token_address(&owner)
+    };
+
+    let account = config.get_account_checked(&token_account_address).await?;
+
+    let state_with_extension = StateWithExtensionsOwned::<Account>::unpack(account.data)?;
+    let token = token_client_from_config(config, &state_with_extension.base.mint, None)?;
+
+    // the amount the user wants to deposit or withdraw, as an f64
+    let maybe_amount =
+        ui_amount.map(|ui_amount| spl_token::ui_amount_to_amount(ui_amount, mint_info.decimals));
+
+    // the amount we will deposit or withdraw, as a u64
+    let amount = if !config.sign_only && instruction_type == ConfidentialInstructionType::Deposit {
+        let current_balance = token
+            .get_account_info(&token_account_address)
+            .await?
+            .base
+            .amount;
```

Comment:
The account was just fetched earlier as `state_with_extension` -- can you just re-use that here?

Reply:
Oops! Fixed!